### PR TITLE
Create full-example.md

### PR DESCRIPTION
**Pull request by rdrpenguin04**

### CODE REVIEW

File: tests/full-example.ds [hunk 1]
```diff
@@ -0,0 +1,42 @@
+extern {
+  fn format(x: str);
+  fn printf(x: str);
+  fn sqrt(x: f64);
+}
+
+struct vec3 {
```
Comment (jorolf) [inline tests/full-example.ds]: ``vec32`` instead of ``vec3``

Comment (rdrpenguin04) [reply tests/full-example.ds]: `vec3` instead of `vec32`

File: tests/full-example.ds [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  vec3(a.x + b.x, a.y + b.y, a.z + b.z);
+}
+
+fn main() {
+  let a = vec3(1, 0, 0);
+  let b = vec3(0, 1, 0);
+  let c = a + b;
+  printf("Length of vector a (%s) is %d", a.toString(), a.length());
+  printf("Length of vector b (%s) is %d", b.toString(), b.length());
+  printf("Length of vector a+b (%s) is %d", c.toString(), c.length());
```
Comment (jorolf) [inline tests/full-example.ds]: to_str

File: tests/full-example.ds [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  fn length() : f32 {
+    f32(sqrt(self.x*self.x + self.y*self.y + self.z*self.z));
+  }
+  
+  fn to_str() : str {
+    format("Vec3 {X: %d, Y: %d, Z: %d}", self.x, self.y, self.z);
+  }
+}
+
+infix fn +(a: vec3, b: vec3) {
```
Comment (jorolf) [inline tests/full-example.ds]: missing return type

Comment (rdrpenguin04) [reply tests/full-example.ds]: In the example put in chat, there wasn't one.

File: tests/full-example.ds [hunk 4]
```diff
@@ -0,0 +1,42 @@
+extern {
+  fn format(x: str) : str;
+  fn printf(x: str);
```
Comment (jorolf) [inline tests/full-example.ds]: 🤔 we need varargs support

Comment (rdrpenguin04) [reply tests/full-example.ds]: But somehow main.ds works out of the box...?

**Pull request merged**